code/modules/projectiles/gun.dm

```diff
@@ -359,4 +363,39 @@
 	if(src && input && !M.stat && in_range(M,src) && !M.restrained() && M.canmove)
 		name = input
 		M << "You name the gun [input]. Say hello to your new friend."
-		return
\ No newline at end of file
+		return
+
+
+/obj/item/weapon/gun/proc/handle_suicide(mob/living/carbon/human/user, mob/living/carbon/human/target, params)
+	if(!ishuman(user) || !ishuman(target))
+		return
+
+	if(semicd)
+		return
+
+	if(user == target)
+		target.visible_message("<span class='warning'>[user] sticks [src] in their mouth, ready to pull the trigger...</span>", \
+			"<span class='userdanger'>You stick [src] in your mouth, ready to pull the trigger...</span>")
+	else
+		target.visible_message("<span class='warning'>[user] points [src] at [target]'s head, ready to pull the trigger...</span>", \
+			"<span class='userdanger'>[user] points [src] at your head, ready to pull the trigger...</span>")
+
+	semicd = 1
+
+	if(!do_mob(user, target, 120) || user.zone_sel.selecting != "mouth")
+		if(user == target && user)
+			user.visible_message("<span class='notice'>[user] decided life was worth living.</span>")
```

Comment:
Because english isn't my first, it isn't even my second and my grammar still sucks a lot. Give me a message for cancelling execution and I will add it.

There is another trouble. You can "decide" against shooting because your target ran away or somebody tased you. Then a feedback message wouldn't fit.